src/realm/object-store/impl/realm_coordinator.cpp

```diff
@@ -967,8 +971,6 @@ void RealmCoordinator::run_async_notifiers()
     // Now that they're at the same version, switch the new notifiers over to
     // the main Transaction used for background work rather than the temporary one
     for (auto& notifier : new_notifiers) {
-        REALM_ASSERT(m_notifier_transaction->get_version_of_current_transaction() ==
-                     notifier->transaction().get_version_of_current_transaction());
```

Comment:
Since we now aren't advancing the notifier's transaction the version won't match, but that's fine as we now support cross-version handover.